plugins/parsers/json/parser.go

```diff
@@ -78,14 +79,31 @@ func (p *JSONParser) parseObject(metrics []telegraf.Metric, jsonOut map[string]i
 			return nil, err
 		}
 
-		timeStr, ok := f.Fields[p.JSONTimeKey].(string)
-		if !ok {
-			err := fmt.Errorf("time: %v could not be converted to string", f.Fields[p.JSONTimeKey])
-			return nil, err
-		}
-		nTime, err = time.Parse(p.JSONTimeFormat, timeStr)
-		if err != nil {
-			return nil, err
+		// unix: epoch is assumed to be in seconds and unquoted. Can have a decimal part.
+		// unix_ms: epoch is assumed to be in milliseconds and unquoted. Cannot have a decimal part.
+		if strings.EqualFold(p.JSONTimeFormat, "unix") || strings.EqualFold(p.JSONTimeFormat, "unix_ms") {
+			timeEpoch, ok := f.Fields[p.JSONTimeKey].(float64) //Using float64 as that is what the JSONFlattener returns
+			if !ok {
+				err := fmt.Errorf("time: %v could not be converted to float64", f.Fields[p.JSONTimeKey])
+				return nil, err
+			}
+
+			if strings.EqualFold(p.JSONTimeFormat, "unix") {
+				timeInt, timeFractional := math.Modf(timeEpoch)
```

Comment:
I've been looking at this method and I'm concerned that it will lose precision for some timestamps.  Here is an example testcase I made:
```
func TestModf(t *testing.T) {
	epoch := 1536107026.123456789
	integer, fractional := math.Modf(epoch)
	tm := time.Unix(int64(integer), int64(fractional*1e9)).UTC()
	require.Equal(t, int64(1536107026123456789), tm.UnixNano())
}
```
```
--- FAIL: TestModf (0.00s)
	Error Trace:	main_test.go:140
	Error:      	Not equal: 
	            	expected: 1536107026123456789
	            	actual  : 1536107026123456716
	Test:       	TestModf
```

That said, this is an issue with the json flattener and while we could delay conversion to float, I don't think it is required now.